### PR TITLE
[feat]: [CDS-71777]: Add admin api to disable triggers

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Describe your changes
Added an admin api so that CSEs or someone with admin priviledge can disable all triggers at a given scope.
Tested on local to check only triggers of given scope were deleted
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51201)

### CODE REVIEW

**Comment by rathodmeetsatish** [comment]
trigger uts

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/NGTriggerService.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ Optional<NGTriggerEntity> get(String accountId, String orgIdentifier, String pro
       String targetIdentifier, String identifier, boolean deleted);
 
   NGTriggerEntity update(NGTriggerEntity ngTriggerEntity, NGTriggerEntity oldNgTriggerEntity);
-
+  TriggerUpdateCount updateEnabledField(String accountIdentifier, String orgIdentifier, String projectIdentifier);
```
Comment (vcalasansh) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/NGTriggerService.java:43]: Maybe we should call this `disableTriggers`, just to make it clear that it is only supposed to disable triggers?

**Review by vcalasansh — COMMENTED** [review]
Added a comment

**Comment by rathodmeetsatish** [comment]
trigger uts

**Comment by rathodmeetsatish** [comment]
trigger ut6,ut0

**Review by vcalasansh — COMMENTED** [review]
> Reviewed 5 of 12 files at r1, 4 of 7 files at r2, 1 of 2 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51201)* status: 14 file(s) to review. 0 of 2 approvals obtained.

**Review by vcalasansh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51201)* status: 14 file(s) to review. 1 of 2 approvals obtained.

**Review by bhavya181 — DISMISSED** [review]

**Comment by rathodmeetsatish** [comment]
trigger review

**Comment by rathodmeetsatish** [comment]
trigger messagecheck

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java` line 563 at r4](https://reviewable.io/reviews/harness/harness-core/51201#-NbDXz7710lCvaoRHmm8:-NbDXz7710lCvaoRHmm9:bqxh904) ([raw file](https://github.com/harness/harness-core/blob/04f785d0528b10b4636b59b505975ebb5181544e/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java#L563)):*
> ```Java
>   public TriggerUpdateCount disableTriggers(String accountIdentifier, String orgIdentifier, String projectIdentifier) {
>     Criteria criteria =
>         Criteria.where(NGTriggerEntityKeys.accountId).is(accountIdentifier).and(NGTriggerEntityKeys.deleted).is(false);
> ```

move deleted query to be at last.

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java` line 570 at r4](https://reviewable.io/reviews/harness/harness-core/51201#-NbDY2E21Ju35eV7JmkU:-NbDY2E21Ju35eV7JmkV:b-3lj6b0) ([raw file](https://github.com/harness/harness-core/blob/04f785d0528b10b4636b59b505975ebb5181544e/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java#L570)):*
> ```Java
>       criteria.and(NGTriggerEntityKeys.projectIdentifier).is(projectIdentifier);
>     }
>     List<NGTriggerEntity> listOfTriggers = ngTriggerRepository.findAll(criteria);
> ```

Use iterators over here and also do batch updates.

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> Reviewed 5 of 12 files at r1, 4 of 7 files at r2, 1 of 2 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51201)* status: Some of the checks are too old. 1 change requests.

**Comment by rathodmeetsatish** [comment]
trigger uts

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51201)* status: 6 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java` line 563 at r4](https://reviewable.io/reviews/harness/harness-core/51201#-NbDXz7710lCvaoRHmm8:-NbO1yhj2pxe9sn3BbBJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/04f785d0528b10b4636b59b505975ebb5181544e/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java#L563)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

move deleted query to be at last. 
</blockquote></details>

Done.
___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java` line 570 at r4](https://reviewable.io/reviews/harness/harness-core/51201#-NbDY2E21Ju35eV7JmkU:-NbO1zE3AOICJ-LVUfBN:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/04f785d0528b10b4636b59b505975ebb5181544e/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java#L570)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Use iterators over here and also do batch updates. 
</blockquote></details>

Done.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51201)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by bhavya181 — APPROVED** [review]

**Comment by rathodmeetsatish** [comment]
trigger review

**Pull request merged**